### PR TITLE
Redux navigation integration functioning

### PR DESCRIPTION
**Pull request by redspanner**

### CODE REVIEW

File: go-for-eat/actions/index.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import { CALL_API } from '../middleware/api';
 import { schema } from 'normalizr';
 
-
 const userSchema = new schema.Entity('users', {}, {idAttribute:'user_id'});
-const eventSchema = new schema.Entity('events', {participants:[userSchema]}, {idAttribute:'event_id'});
+const userArraySchema = [userSchema];
```
Comment (arol) [inline go-for-eat/actions/index.js:5]: Ract Navigation and some bonus tracks!

File: go-for-eat/navigation/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  Home: {
+    screen: NavigatorWithRootScreen(Home),
+    navigationOptions:{header:()=>null}
+  },
+
+  User: {
+    screen: NavigatorWithRootScreen(User),
+    navigationOptions:{header:()=>null}
+  },
+  CreateEvent: {
```
Comment (arol) [inline go-for-eat/navigation/index.js]: Sorry, doesn't it mean that CreateEvent is now part of the `HomeStack` so it will be pushed when presented?

Comment (arol) [reply go-for-eat/navigation/index.js]: I imagined it to be something like
`NavigatorWithModal(NavigatorWithRootScreen(MainNavigator), CreateEvent)`

File: go-for-eat/screens/Login/Login.js [hunk 3]
```diff
@@ -104,6 +102,7 @@ const mapStateToProps = (state) => ({
 const mapDispatchToProps = (dispatch) => ({
   setUser: user => dispatch(setUser(user)),
   serverAuth: (data) => dispatch(loginUser(data)),
+  goHome: () => dispatch({type:'Home'})
```
Comment (arol) [inline go-for-eat/screens/Login/Login.js]: I suggest naming the action to something more specific like `'navigation.home'`

**Review by arol — APPROVED** [review]

File: go-for-eat/components/NavBar/NavBar.js [hunk 4]
```diff
@@ -1,36 +1,103 @@
...
+    this.props.navigate('Profile');
+  }
 
+  handleCreate = () => {
+    this.props.navigate('CreateEvent');
   }
 
-  renderNavBackButton = () => {
+  handleLogout = async () => {
+    new Promise((resolve,reject) => {
```
Comment (arol) [inline go-for-eat/components/NavBar/NavBar.js]: Why? :(
This will resolve immediately.

File: go-for-eat/components/NavBar/NavBar.js [hunk 5]
```diff
@@ -1,36 +1,103 @@
...
+      >
+        <Image
+          style={style.navbar_icon}
+          source={navBack}
+        />
+      </TouchableOpacity>
+    )
   }
 
+  // renderButton = (onPress, icon) => {
```
Comment (arol) [inline go-for-eat/components/NavBar/NavBar.js]: Remove unused code.

File: go-for-eat/components/NavBar/NavBar.js [hunk 6]
```diff
@@ -43,24 +110,61 @@ class NavBar extends Component {
     )
   }
 
+
   render () {
+
+    // const buttons={}
```
Comment (arol) [inline go-for-eat/components/NavBar/NavBar.js]: Remove unused/commented code.

File: go-for-eat/localStorage.js [hunk 7]
```diff
@@ -12,10 +12,20 @@ export const loadState = () => {
...
     Expo.SecureStore.setItemAsync('state', serializedState);
   } catch (err) {
     console.log(err);
   }
 };
+
+export const logoutStorage = () => {
+  try {
+    const serializedState = JSON.stringify({});
+    Expo.SecureStore.setItemAsync('state', serializedState);
```
Comment (arol) [inline go-for-eat/localStorage.js]: If you return this, you'll make logoutStorage to be an asynchronous operation.
And the try catch in the level above, so next to the business logic.

**Review by arol — CHANGES_REQUESTED** [review]

File: go-for-eat/components/NavBar/NavBar.js (hunk 4 above)
Comment (redspanner) [reply go-for-eat/components/NavBar/NavBar.js]: :(

File: go-for-eat/components/NavBar/NavBar.js [hunk 8]
```diff
@@ -30,12 +29,11 @@ class NavBar extends Component {
     this.props.navigate('CreateEvent');
   }
 
-  handleLogout = async () => {
-    new Promise((resolve,reject) => {
-      resolve(logoutStorage());
-    })
-    .then(()=>this.props.logoutState())
-    .then(()=>this.props.navigate('Login'))
+  handleLogout = () => {
```
Comment (arol) [inline go-for-eat/components/NavBar/NavBar.js]:
```js
handleLogout = async () => {
    const serializedState = JSON.stringify({});
    await Expo.SecureStore.setItemAsync('state', serializedState);
    await this.props.logoutState(); // if async
    this.props.navigate('Login');
}
```

**Review by arol — APPROVED** [review]

**Review by div-Leo — APPROVED** [review]
just for this time

**Pull request merged**